milestones/api.py

```diff
@@ -268,25 +284,36 @@ def add_course_content_milestone(course_key, content_key, relationship, mileston
         course_key=course_key,
         content_key=content_key,
         relationship=relationship,
-        milestone=milestone)
+        milestone=milestone,
+        requirement=requirement
```

Comment:
Yes, that was the idea of this field. The current gating use case does not call for anything more than a primitive value, but perhaps it does make sense for the milestones API to present this field as a dict to clients which is serialized to json on write. I will change the name of this field to requirements and make the API change. Thanks.